public/app/features/playlist/api.ts

```diff
@@ -11,71 +11,132 @@ import { dispatch } from 'app/store/store';
 
 import { DashboardQueryResult, getGrafanaSearcher, SearchQuery } from '../search/service';
 
-import { Playlist, PlaylistItem, KubernetesPlaylist, KubernetesPlaylistList, PlaylistAPI } from './types';
+import { Playlist, PlaylistItem, PlaylistAPI } from './types';
 
-export async function createPlaylist(playlist: Playlist) {
-  await withErrorHandling(() => getBackendSrv().post('/api/playlists', playlist));
+class LegacyAPI implements PlaylistAPI {
+  async getAllPlaylist(): Promise<Playlist[]> {
+    return getBackendSrv().get<Playlist[]>('/api/playlists/');
+  }
+
+  async getPlaylist(uid: string): Promise<Playlist> {
+    const p = await getBackendSrv().get<Playlist>(`/api/playlists/${uid}`);
+    await migrateInternalIDs(p);
+    return p;
+  }
+
+  async createPlaylist(playlist: Playlist): Promise<void> {
+    await withErrorHandling(() => getBackendSrv().post('/api/playlists', playlist));
+  }
+
+  async updatePlaylist(playlist: Playlist): Promise<void> {
+    await withErrorHandling(() => getBackendSrv().put(`/api/playlists/${playlist.uid}`, playlist));
+  }
+
+  async deletePlaylist(uid: string): Promise<void> {
+    await withErrorHandling(() => getBackendSrv().delete(`/api/playlists/${uid}`), 'Playlist deleted');
+  }
 }
 
-export async function updatePlaylist(uid: string, playlist: Playlist) {
-  await withErrorHandling(() => getBackendSrv().put(`/api/playlists/${uid}`, playlist));
+interface KubernetesPlaylistList {
```

Comment:
Nit to unify with the rest of the file type/class names 

```suggestion
interface K8sPlaylistList {
```